pkg/cleaner/exec_process_cleaner.go

```diff
@@ -0,0 +1,14 @@
+package cleaner
+
+import (
+	"context"
+	"os/exec"
+)
+
+// NewExecProcessCleaner creates a new Cleaner that exec's an command on the
+// system and expects a 0 exit code.
+func NewExecProcessCleaner(command string, args []string) Cleaner {
+	return func(ctx context.Context) error {
+		return exec.CommandContext(ctx, command, args...).Run()
```

Comment:
I was going to ask you: Why can't we let NewExecProcessCleaner() take an *exec.Cmd, so that we can clone it as part of this function? That way consumers of this Cleaner may also override the working directory, environment variables, SysProcAttr, etc. It turns out that exec.Cmd is a bit of a mess:

- It contains both inputs (environment variables, arguments) and outputs (process state, LookPath errors),
- There is no way to override the `ctx` after it's been created.

So yeah. Let's stick to this approach. :-/

One recommendation I have, though. Could you change this to something like this?
```go
if err := exec.CommandContext(ctx, command, args...).Run(); err != nil {
     return util.StatusWrapWithCode(err, codes.Internal, "Failed to run cleaning command")
}
return nil
```
Otherwise it may not be obvious from the logs that any failures are caused by this feature.

Reply:
Yes, thats a great callout! I had not considered this.